models/labels/trader_dex_diversity/labels_trader_dex_diversity.sql

```diff
@@ -0,0 +1,33 @@
+{{
+    config(
+        alias='trader_dex_diversity',
+        post_hook='{{ expose_spells(\'
+        ["ethereum", "fantom", "arbitrum", "avalanche_c", "gnosis", "bnb", "optimism", "polygon"]\', 
+        "sector", 
+        "labels", 
+        \'["gentrexha", "Henrystats"]\') }}'
+    )
+}}
+
+with
+ trader_dex_diversity as (
+    select blockchain,
+           count(distinct project) as dex_diversity,
+           taker                   as address
+    from (select blockchain, taker, project
+          from {{ ref('dex_aggregator_trades') }}
+          union all
+          select blockchain, taker, project
+          from {{ ref('dex_trades') }})
+    group by taker, blockchain 
+ )
+
+select array(blockchain)                                   as blockchain,
+       address,
+       concat('Number of DEXs traded on: ', dex_diversity) as name,
+       "trader_dex_diversity"                              AS category,
+       "gentrexha"                                         AS contributor,
+       "query"                                             AS source,
+       timestamp('2022-12-15')                             as created_at,
+       now()                                               as updated_at
+from trader_dex_diversity
```

Comment:
Do you want to change `array(blockchain)` to `collect_set(blockchain)`?
https://docs.databricks.com/sql/language-manual/functions/collect_set.html
You can also group by `address` here.
cc. @gentrexha

Reply:
it was originally collect_set(blockchain) but we(with @gentrexha) discussed to make the spells e.g dex diversity grouped by blockchain too so the data is unique for an address that's active across multiple chains. 

So the diversity on ethereum would be different as the one on arbitrum if the trader trades differently.


There have actually been another spell merged (trader age and these ones are similar). cc @soispoke